### PR TITLE
Fix rect drawing with width

### PR DESCRIPTION
**Pull request by MightyJosip**
Not much explanation needed, this is a fix for #2862

### CODE REVIEW

**Comment by ankith26** [comment]
whoops, seems like the manylinux1 builds are not happy. Might have to do something with the fact that `centos5` uses an old gcc

**Review by MyreMylar — APPROVED** [review]
Tested it locally and it does what it says on the tin.

Looks good. I like the new, properly rectangular, corners.

**Review by illume — CHANGES_REQUESTED** [review]
Thanks. Lgtm too.

A test case would be nice if possible. If you don’t have time, let me know… I can whip one up?

File: src_c/draw.c [hunk 1]
```diff
@@ -67,21 +67,23 @@ static void
...
 draw_round_rect(SDL_Surface *surf, int x1, int y1, int x2, int y2, int radius,
                 int width, Uint32 color, int top_left, int top_right,
                 int bottom_left, int bottom_right, int *drawn_area);
 
 // validation of a draw color
 #define CHECK_LOAD_COLOR(colorobj)                                         \
-    if (PyLong_Check(colorobj))                                            \
-        color = (Uint32)PyLong_AsLong(colorobj);                           \
+    if (PyInt_Check(colorobj))                                             \
+        color = (Uint32)PyInt_AsLong(colorobj);                            \
```
Comment (ankith26) [inline src_c/draw.c:86]: This change breaks stuff, because the `PyInt_*` macros no longer exist

File: src_c/draw.c [hunk 2]
```diff
@@ -511,7 +509,7 @@ arc(PyObject *self, PyObject *arg, PyObject *kwargs)
 {
     pgSurfaceObject *surfobj = NULL;
     PyObject *colorobj = NULL, *rectobj = NULL;
-    SDL_Rect *rect = NULL, temp;
+    GAME_Rect *rect = NULL, temp;
```
Comment (ankith26) [inline src_c/draw.c:512]: use `SDL_Rect` in place of `GAME_Rect`

File: src_c/draw.c [hunk 3]
```diff
@@ -2473,21 +2360,21 @@ MODINIT_DEFINE(draw)
...
     }
     import_pygame_rect();
     if (PyErr_Occurred()) {
-        return NULL;
+        MODINIT_ERROR;
     }
     import_pygame_surface();
     if (PyErr_Occurred()) {
-        return NULL;
+        MODINIT_ERROR;
```
Comment (ankith26) [inline src_c/draw.c:2375]: do `return NULL;` instead of `MODINIT_ERROR;`

**Review by ankith26 — COMMENTED** [review]
Whoops looks like a merge conflict fix went bad here 😅 
Left some reviews that should be fixed
Also once your PR is done you should do a `python3 setup.py format` to fix lint

**Comment by MightyJosip** [comment]
Closing because this branch is in a weird state after rebasing, continued in #2927

**Pull request closed without merging**